### ISSUE TITLE
Blank Screen on Callback

### ISSUE THREAD

**Issue by Snugug** [Herohtar/netlify-cms-oauth-firebase#5]
Hey!

I've got this set up per the instructions, including adding billing for external requests. I've got it launched, and I can get to the auth screen, but at the callback I'm getting a blank screen and it's not bringing me back to my app. Is there something in the instructions that I'm missing that would allow this to come full circle for me?

Thanks!

**Comment by Herohtar** [comment]
I can't make any guesses at what the problem might be without some details. Can you check your Firebase Function logs and tell me whether there are any errors related to the OAuth function or if it is executing successfully? Also, what is the URL of the blank page you are getting? Finally, if you view the source of the blank page, is there anything there, and if so, what is it? Thanks.

**Comment by Snugug** [comment]
Hey, sorry I wasn't clear enough initially!

The URL is `/oauth/callback?code=CODE&state=STATE` and the body is 

```html
<!doctype html><html><body><script>
  (function() {
    function receiveMessage(e) {
      console.log("receiveMessage %o", e)
      window.opener.postMessage(
        'authorization:github:success:{"token":"TOKEN","provider":"github"}',
        e.origin
      )
      window.removeEventListener("message",receiveMessage,false);
    }
    window.addEventListener("message", receiveMessage, false)
    console.log("Sending message: %o", "github")
    window.opener.postMessage("authorizing:github", "*")
    })()
  </script></body></html>
```

There are no errors in my logs, only two successful function executions, the first being resulting in `302` and the second resulting in a `200` status code.

**Comment by Herohtar** [comment]
Interesting. Everything appears to be in order there... what browser and OS are you using?

**Comment by vella-nicholas** [comment]
@Snugug I am having a similar issue. Are you hosting Netlify locally?

**Comment by keithazzopardi** [comment]
@Herohtar we are also having a similar issue....after successfully authenticating on github, on the callback page we have the log `Sending message: "github"`. However there is no other logs in the parent window nor in the child popup. This behaviour is happening on both Firefox and Chrome browsers. On the function there is no errors.

It seems to me that there is some redirection issue from the popup to the parent. Is it possible?

**Comment by Herohtar** [comment]
I am unable to reproduce the issue in both Firefox 70.0b7 and Chrome 77 on Windows 10. The redirect for the OAuth login on @Snugug's site works as expected for me (though I can't completely login since I don't have access to his repo).

@keithazzopardi Seeing `Sending message: "github"` means the OAuth portion has completed successfully and the Firebase function has returned the final page. After that, the popup calls `window.opener.postMessage("authorizing:github", "*")` to send a signal back to the parent, which should respond in turn, causing another console message in the popup before it sends the OAuth tokens to the parent page. Since you do not see any other logs, that means something is failing with the communication between the parent page and the popup.

Either something isn't set up correctly on your admin login page, or you have some browser setting or addon that is interfering with the JavaScript communication.

**Comment by Herohtar** [comment]
I'm closing this since I can't reproduce it and there hasn't been any more information.

**Comment by JaySunSyn** [comment]
Be aware that you need upgrade your firebase plan to a paid version e.g. `Blaze
Pay as you go` in order to allow outgoing requests.

**Comment by CurtisMac** [comment]
I've just started having a similar issue. I set everything up a while back and it was working fine. I haven't logged in for a while, so not sure when the issue started, I only noticed it today. Like Snugug, I am getting the expected response printed to the console and the url is correct. No errors are being logged in Firebase. 
The error is not consistent. If I am running netlify on local host, I can log in no issues and everything works. When I try to log into the version of my site being hosted on Firebase, it works if I am in incognito mode in Chrome (Version 79.0.3945.117) & Opera (66.0.3515.27), but not in a normal window. I've tried clearing out the browser cache, but it still only works in incognito mode. 
Everything works fine in Firefox. So I suspect there is some conflict within the browser.
I'm not sure how to go about debugging the issue.
